auto-value-cursor/src/main/java/com/gabrielittner/auto/value/contentvalues/AutoValueContentValuesExtension.java

```diff
@@ -91,21 +95,34 @@ private MethodSpec createToContentValuesMethod(AutoValueExtension.Context contex
                     continue;
                 }
 
-                builder.addStatement("$1T $2LValues = $3T.$4N($2L())", CONTENT_VALUES, name,
+                writeMethod.addStatement("$1T $2LValues = $3T.$4N($2L())", CONTENT_VALUES, name,
                         TypeName.get(factoryTypeMirror), method.getSimpleName().toString());
-                builder.addStatement("if ($1LValues != null) values.putAll($1LValues)", name);
+                writeMethod.addStatement("if ($1LValues != null) values.putAll($1LValues)", name);
             } else if (getCursorMethod(type) != null) {
+                boolean isNullable = hasAnnotationWithName(element, NULLABLE);
                 String columnName = getColumnName(element);
-                if (columnName == null) columnName = entry.getKey();
-                builder.addStatement("values.put($S, $L())", columnName, name);
+                columnName = columnName != null ? columnName : name;
+                CodeBlock writeCode = CodeBlock.builder()
+                        .addStatement("values.put($S, $L())", columnName, name)
+                        .build();
+                if (isNullable) {
+                    writeCode = CodeBlock.builder()
+                            .beginControlFlow("if ($L() == null)", name)
+                            .addStatement("values.putNull($S)", columnName)
```

Comment:
I don't think you need to bother with this. `put` accepts `null` values (it's just backed by a `Map`).